### PR TITLE
[MINI-5295] UI freezes when inserting big batch of data

### PR DESCRIPTION
**Pull request by laubengaier**
# Description
UI freezes when inserting big batch of data
- Putting `set` and `remove` code into background threats
- Refactor Unit Tests from Nimble to XCTests
- additional unit tests
- vacuum after `remove`

MiniApps should handle blocking the ui themselves

## Links
~~https://jira.rakuten-it.com/jira/browse/MALI-1086~~
https://jira.rakuten-it.com/jira/browse/MINI-5295

# Checklist
- [x] I have read the [contributing guidelines](CONTRIBUTING.md).
- [x] I have completed the [SDK Development Learning Path](CONTRIBUTING.md#SDK-Development-Learning-Path) (if submitting a feature PR)
- [x] I wrote/updated tests for new/changed code
- [x] I removed all sensitive data before every commit, including API endpoints and keys
- [ ] I ran `fastlane ci` without errors

### CODE REVIEW

File: Sources/Classes/core/SecureStorage/MiniAppSecureStorage.swift [hunk 1]
```diff
@@ -67,19 +67,22 @@ class MiniAppSecureStorage: MiniAppSecureStorageDelegate {
...
-                case .success:
-                    completion?(.success(true))
-                case let .failure(error):
-                    completion?(.failure(error))
-                }
-            })
-        } catch {
-            completion?(.failure(.storageIOError))
-            return
+        DispatchQueue.global(qos: .userInitiated).async {
```
Comment (rleojoseph) [inline Sources/Classes/core/SecureStorage/MiniAppSecureStorage.swift:70]: Shouldn't we do the same for Delete as well ?

Comment (laubengaier) [reply Sources/Classes/core/SecureStorage/MiniAppSecureStorage.swift]: yeah true, fixed it

**Review by rleojoseph — APPROVED** [review]
Looks good to me :+1:

**Review by munir-rakuten — APPROVED** [review]
LGTM

**Pull request merged**